aquarius/events/util.py

```diff
@@ -47,6 +47,28 @@ def get_network_name():
     return network_name
 
 
+def get_network_name_by_chain_id(chain_id: int) -> str:
+    """Return the contract address with the given name and chain id"""
+    with open(get_address_file(), "r") as address_json:
+        addresses = json.load(address_json)
+    return next(
+        network_name
+        for (network_name, network_values) in addresses.items()
+        if network_values["chainId"] == chain_id
+    )
+
+
+def get_defined_block(chain_id: int):
+    """Retrieves the block either from envvar, either from address.json file."""
+    with open(get_address_file(), "r") as address_file:
+        addresses = json.load(address_file)
+    return (
+        int(os.getenv("BFACTORY_BLOCK"))
+        if "BFACTORY_BLOCK" in os.environ
+        else addresses[get_network_name_by_chain_id(chain_id)]["startBlock"]
```

Comment:
you don't need the network name. You can retrieve the start block by chain_id directly.